h5io/tests/test_io.py

```diff
@@ -499,3 +500,105 @@ def test_state_python_version_error(tmp_path):
         overwrite="update",
         use_state=True,
     )
+
+
+class Singleton(ABCMeta):
+    """
+    Copied from pyiron_base.
+
+    https://github.com/pyiron/pyiron_base/blob/33910343e5e6d4c8bbb5f2522ad6714ec5184ff5/pyiron_base/interfaces/singleton.py#L23
```

Comment:
I guess the original reference is http://stackoverflow.com/questions/6760685/creating-a-singleton-in-python still I am not sure if we need these references in the age of ChatGPT when we copy five lines. I would remove it